### PR TITLE
Add storybook for UI components, migrate Remix to vite

### PR DESCRIPTION
**Pull request by andrelandgraf**
## Summary (1-2 sentences)

This PR adds [Storybook](https://storybook.js.org/) to showcase our reusable UI components. It further migrates Remix to use vite.

### Motivation

Storybook allows us to browse and visually compare our UI components quickly. This should improve the discoverability and documentation of our UI library components. Storybook uses Vite to compile our React components. Since Vercel now supports Remix on Vite as well, we also migrated Remix to Vite, which was, luckily, fairly easy!

Note that we could have split the Remix Vite migration and Storybook addition into two separate PRs for easier review. However, since we paired on this, I think it's fine! :) 

### Sources (links that helped to resolve these challenges.) (optional)

- Remix on Vercel docs: https://vercel.com/docs/frameworks/remix#vercel-vite-preset
- Open Remix RFC to better support Storybook: https://github.com/remix-run/remix/discussions/2481

### Usage

We can now run `npm run storybook` to visit the Storybook UI on port 6006 to review our existing UI components.

### Next steps

- Add all UI components to Storybook
- Host Storybook on GitHub Pages so we can view it without first running it locally

## Screenshots/Recordings (if applicable)

Screenshot 2024-03-11 at 9 48 46 PM

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -1,17 +1,10 @@
 // eslint-disable-next-line @typescript-eslint/no-var-requires
-const config = require('./.prettierrc');
+const config = require('./.prettierrc.cjs');
```
Comment (andrelandgraf) [inline .eslintrc.cjs:2]: We switched to ESM for our application. To opt out of this, we can mark prettier and ESLint as `cjs` (common JS) files as they don't seem to support ESM in our current set up.

File: .storybook/main.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "@storybook/addon-links",
+    "@storybook/addon-essentials",
+    "@chromatic-com/storybook",
+    "@storybook/addon-interactions"
+  ],
+  "framework": {
+    "name": "@storybook/react-vite",
+    "options": {
+      builder: {
+        viteConfigPath: 'storybook.vite.config.ts'
```
Comment (andrelandgraf) [inline .storybook/main.ts:19]: The Remix Vite plug-in doesn't work with Storybook. We use a custom storybook vite config as a workaround.

File: package.json [hunk 3]
```diff
@@ -1,26 +1,29 @@
 {
   "private": true,
   "sideEffects": false,
+  "type": "module",
```
Comment (andrelandgraf) [inline package.json:4]: ES modules now!

**Review by eiffelwong1 — APPROVED** [review]
lgtm

**Review by shami-dev — APPROVED** [review]
Amazing! 🤩

**Review by tonydangblog — APPROVED** [review]

**Review by eiffelwong1 — APPROVED** [review]
lgtm

**Pull request merged**